### PR TITLE
trying to populate index() for result of outer product, but the index…

### PR DESCRIPTION
**Pull request by wdednam**
…() objects always seem to be empty regardless of whether I load sparse or dense tensors?

---
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/253)

### CODE REVIEW

File: engine/minskyTensorOps.cc [hunk 1]
```diff
@@ -393,6 +399,15 @@ namespace minsky
       hc.xvectors.insert(hc.xvectors.begin(), xv1.begin(), xv1.end());           
       cachedResult.hypercube(move(hc));
         
+      set<size_t> newIdx;
+      for (auto& i: arg1->index())
+        for (auto& j: arg2->index()) 
+        {
+            newIdx.insert(i+arg1->index().size()*j);
```
Comment (highperformancecoder) [inline engine/minskyTensorOps.cc:406]: This doesn't look right. The new index should be i+arg1->hypercube().numElements()*j, but obviously move the numElements() call outside the loop.

File: engine/minskyTensorOps.cc [hunk 2]
```diff
@@ -368,17 +368,23 @@ namespace minsky
...
-		}
-	  }
+      //if (cachedResult.index().empty())  	
+       for (size_t i=0; i< m; i++)
+       {
+         auto v1=(*arg1)[i];  			
+         for (size_t j=0; j< n; j++) 
+         {
+             auto v2=(*arg2)[j];			
+             if (!isnan(v1) && !isnan(v2)) cachedResult[i+j*m]=v1*v2;
```
Comment (highperformancecoder) [inline engine/minskyTensorOps.cc:380]: This should be fine. Note the test for isnan is not necessary, a nan multiplied by anything else is still a nan.

File: engine/minskyTensorOps.cc (hunk 1 above)
Comment (wdednam) [reply engine/minskyTensorOps.cc]: Thanks, done!

File: engine/minskyTensorOps.cc (hunk 2 above)
Comment (wdednam) [reply engine/minskyTensorOps.cc]: Ok, I got rid of the conditional.

**Pull request merged**